#VRML_SIM R2021b utf8
WorldInfo {
  info [
    "GANKENKUN robot."
    "The GANKENKUN robot simulation model"
  ]
  title "GANKENKUN"
  basicTimeStep 8
  optimalThreadCount 8
  physicsDisableTime 0.1
  physicsDisableLinearThreshold 0.1
  physicsDisableAngularThreshold 0.1
  contactProperties [
    ContactProperties {
      material1 "grass"
      coulombFriction [
        0.5
      ]
      softCFM 0.03
    }
    ContactProperties {
      material1 "grass"
      material2 "robocup soccer ball"
      coulombFriction [
        0.5
      ]
      bounce 0.76
      softCFM 0.05
    }
    ContactProperties {
      material2 "robocup soccer ball"
      bounce 0.76
    }
  ]
}
Viewpoint {
  orientation 0.6719791180295076 -0.49317350564460494 -0.552470775935249 1.8510428047885563
  position -2.5375047220358824 -0.9166711516828123 0.6010308394649483
}
TexturedBackground {
  texture "stadium_dry"
}
TexturedBackgroundLight {
  texture "stadium_dry"
}

DEF BLUE_PLAYER1 GankenKun_simple {
  translation -0.3 0 0.450
  rotation 0 0 1 0
  name "blue player 1"
  jerseyTexture "textures/GankenKun_blue1.png"
  jerseyColor 0, 0, 1
  controller "GankenKun_walking"
  supervisor TRUE
}

DEF BLUE_PLAYER2 GankenKun_simple {
  translation -2 -1 0.450
  rotation 0 0 1 0
  name "blue player 2"
  jerseyTexture "textures/GankenKun_blue2.png"
  jerseyColor 0, 0, 1
  controller "GankenKun_walking"
  supervisor TRUE
}
  
DEF BLUE_PLAYER3 GankenKun_simple {
  translation -2 1 0.450
  rotation 0 0 1 0
  name "blue player 3"
  jerseyTexture "textures/GankenKun_blue3.png"
  jerseyColor 0, 0, 1
  controller "GankenKun_walking"
  supervisor TRUE
}
  
DEF RED_PLAYER1 GankenKun_simple {
  translation 1 0 0.450
  rotation 0 0 1 3.14
  name "red player 1"
  jerseyTexture "textures/GankenKun_red1.png"
  jerseyColor 1, 0, 0
  controller "GankenKun_walking"
}
DEF RED_PLAYER2 GankenKun_simple {
  translation 2 -1 0.450
  rotation 0 0 1 3.14
  name "red player 2"
  jerseyTexture "textures/GankenKun_red2.png"
  jerseyColor 1, 0, 0
  controller "GankenKun_walking"
}
DEF RED_PLAYER3 GankenKun_simple {
  translation 2 1 0.450
  rotation 0 0 1 3.14
  name "red player 3"
  jerseyTexture "textures/GankenKun_red3.png"
  jerseyColor 1, 0, 0
  controller "GankenKun_walking"
}
RobocupSoccerField {
  size "kid"
}
DEF BALL RobocupSoccerBall {
  translation 0 0 0.1
  size 1
}

